### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by thedrakeum**
Implementing clang tidy modernize* and readability* checks. (excluded modernize-use-trailing-return-type)
Updated build warning level to W4 and fixed any warnings.
Refactored MemoryGame a bit.

### CODE REVIEW

File: EAD_Arcade/EAD_Arcade.vcxproj [hunk 1]
```diff
@@ -76,6 +76,8 @@
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <LinkIncremental>true</LinkIncremental>
+    <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
+    <ClangTidyChecks>modernize*,readability*,-modernize-use-trailing-return-type</ClangTidyChecks>
```
Comment (cdances) [inline EAD_Arcade/EAD_Arcade.vcxproj:80]: We should make sure that all configurations / platforms have Clang turned on and warning level turned to 4.

File: EAD_Arcade_Lib/EAD_Arcade_Lib.vcxproj [hunk 2]
```diff
@@ -95,6 +95,8 @@
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <LinkIncremental>true</LinkIncremental>
+    <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
+    <ClangTidyChecks>modernize*,readability*,-modernize-use-trailing-return-type</ClangTidyChecks>
```
Comment (cdances) [inline EAD_Arcade_Lib/EAD_Arcade_Lib.vcxproj:99]: Same as before: We should make sure that all configurations / platforms have Clang turned on and warning level turned to 4.

File: EAD_Arcade_Lib/source/WhackAMole.cpp [hunk 3]
```diff
@@ -3,9 +3,10 @@
 
 using namespace std;
```
Comment (cdances) [inline EAD_Arcade_Lib/source/WhackAMole.cpp:4]: Remove "using namespace std;" from .cpp file? Seems like a goal was to remove this everywhere possible?

File: EAD_Arcade_Lib_Unit_Tests/EAD_Arcade_Lib_Unit_Tests.vcxproj [hunk 4]
```diff
@@ -36,6 +36,10 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <IncludePath>../EAD_Arcade/header;$(IncludePath)</IncludePath>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
+    <ClangTidyChecks>modernize*,readability*,-modernize-use-trailing-return-type</ClangTidyChecks>
+  </PropertyGroup>
```
Comment (cdances) [inline EAD_Arcade_Lib_Unit_Tests/EAD_Arcade_Lib_Unit_Tests.vcxproj:42]: See comment on enabling clang / warnings for all configurations.

Comment (thedrakeum) [reply EAD_Arcade_Lib_Unit_Tests/EAD_Arcade_Lib_Unit_Tests.vcxproj:44]: Good point.  Pushed up that change.

File: EAD_Arcade_Lib/EAD_Arcade_Lib.vcxproj (hunk 2 above)
Comment (thedrakeum) [reply EAD_Arcade_Lib/EAD_Arcade_Lib.vcxproj:101]: Good point.  Pushed up that change.

File: EAD_Arcade/EAD_Arcade.vcxproj (hunk 1 above)
Comment (thedrakeum) [reply EAD_Arcade/EAD_Arcade.vcxproj:82]: Good point.  Pushed up that change.

File: EAD_Arcade_Lib/source/WhackAMole.cpp (hunk 3 above)
Comment (thedrakeum) [reply EAD_Arcade_Lib/source/WhackAMole.cpp:4]: Yeah. After watching a bunch of the lectures and seeing it online it seems that "using namespace std" is considered a bad practice.  Best practice is to use std::.  Clang-Tidy will actually complain if you use string vs std::string.

File: EAD_Arcade_Lib/source/MemoryGame.cpp [hunk 5]
```diff
@@ -1,72 +1,77 @@
...
 
-    int i;
-    string sValue;
-    bool bIsNumeric;
+void MemoryGame::play()
+{
+    std::cout << "Playing the Memory Game" << std::endl;
+    std::cout << "How many numbers do you remember?" << std::endl;
 
-    srand((unsigned int) time(NULL));
```
Comment (simoneml-ead) [inline EAD_Arcade_Lib/source/MemoryGame.cpp:25]: Why was this removed?

Comment (thedrakeum) [reply EAD_Arcade_Lib/source/MemoryGame.cpp:25]: I just moved it to the constructor.  Since it only needs to be initialized once.

Comment (simoneml-ead) [reply EAD_Arcade_Lib/source/MemoryGame.cpp:25]: OK. I didn't see that you moved it to the constructor. That's fine.
